frontend/src/use-cases/songs/screens/edit-song/EditSong.screen.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect, useState } from "react";
+import { DigitLoading, useGammaStatus } from "@cthit/react-digit-components";
+import { useParams } from "react-router-dom";
+import { getSong } from "../../../../api/songs/get.songs.api";
+import useAdmin from "../../../../common/hooks/use-admin";
+import InsufficientAccess from "../../../../common/components/insufficient-access";
+import EditSongForm from "./components/edit-song-form";
+import FourZeroFour from "../../../../common/components/four-zero-four";
+import SongFormContainer from "../../components/song-form-container";
+import FormatSongInstructions from "../../components/format-song-instruction";
+import FiveZeroZeroComponent from "../../../../common/components/five-zero-zero";
+
+const EditSong = () => {
+    let { song_id } = useParams();
+    const [faultySongId, setFaultySongId] = useState(false);
+    const [songToEdit, setSongToEdit] = useState(null);
+    const [somethingWrong, setSomethingWrong] = useState(false);
+
+    useEffect(() => {
+        getSong(song_id)
+            .then(res => {
+                setSongToEdit(res.data.data.song);
+                setFaultySongId(false);
+            })
+            .catch(err => {
+                setFaultySongId(true);
+            });
+    }, [song_id]);
+
+    const admin = useAdmin();
+    const [loading] = useGammaStatus();
+    if (!loading && !admin) {
+        return <InsufficientAccess />;
+    }
+
+    if (faultySongId) {
+        return <FourZeroFour />;
+    }
+
+    if (somethingWrong) {
+        return <FiveZeroZeroComponent />;
+    }
+
+    return (
+        <>
+            <DigitLoading
+                loading={songToEdit === null}
+                margin={{ left: "auto", right: "auto", top: "32px" }}
+            />
+            {songToEdit !== null && (
+                <SongFormContainer>
+                    <div />
```

Comment:
Why?

Reply:
I suspect that it can be done with the css-effect Grid-column but I fail to make the css apply to get the desired results.

Reply:
i rewrote the structure of that page and this isn't a problem anymore